src/olympia/versions/models.py

```diff
@@ -168,6 +169,32 @@ def auto_approvable(self):
         )
         return qs
 
+    def should_have_due_date(self, not_=False):
+        return getattr(self, 'exclude' if not_ else 'filter')(
```

Comment:
I've a feeling there's a cleaner way to do this but it works.

Reply:
How about something like:
```python
method = getattr(self, 'exclude' if not_ else 'filter')
is_theme = Q(addon__type__in=amo.GROUP_TYPE_THEME)
has_auto_approval_delayed = Q(addon__reviewerflags__auto_approval_delayed_until__isnull=False)
requires_manual_listed_approval_and_is_listed = Q(
    Q(addon__reviewerflags__auto_approval_disabled=True)
    | Q(
        addon__reviewerflags__auto_approval_disabled_until_next_approval=True  # noqa
    )
    | Q(
        addon__promotedaddon__group_id__in=(
            g.id for g in PRE_REVIEW_GROUPS
        )
    ),
    channel=amo.CHANNEL_LISTED,
)
requires_manual_unlisted_approval_and_is_unlisted = Q(
    Q(addon__reviewerflags__auto_approval_disabled_unlisted=True)
    | Q(
        addon__reviewerflags__auto_approval_disabled_until_next_approval_unlisted=True  # noqa
    ),
    channel=amo.CHANNEL_UNLISTED,
),
return method(
    Q(file__status=amo.STATUS_AWAITING_REVIEW)
    & Q(reviewerflags__pending_rejection__isnull=True)
    & Q(is_theme | has_auto_approval_delayed | requires_manual_listed_approval_and_is_listed)
)
```

?